# Hypothesis testing problems

## Exercise 1

You are a nutritionist researching two different types of diets to see if there is a significant difference in weight loss after one month. You choose two random groups of people; one group follows the first diet, and the other follows the second. At the end of the month, you record the weight loss (in kg) for each person:

| Diet 1 | Diet 2 |
|:-------|:-------|
| 2.0 | 3.0 |
| 2.5 | 3.2 |
| 3.0 | 3.1 |
| 2.8 | 2.9 |
| 2.3 | 2.8 |
| 2.7 | 3.0 |
| 2.5 | 3.2 |

With these data, it seeks to answer the following question: Is there a significant difference in average weight loss between people who followed the first diet and those who followed the second diet?

To draw conclusions, follow the points below:

- State the hypothesis: null and alternative hypothesis.
- Perform the test to test the hypothesis. You can use a Student's t-test.
- Analyze the conclusions.

Null Hypothesis: There is no significant difference in average weight loss between people following Diet 1 and people following Diet 2.

Alternative Hypothesis: There is a significant difference in average weight loss between people following Diet 1 and people following Diet 2.

In [2]:
# TODO
import numpy as np
from scipy import stats

diet1 = np.array([2.0, 2.5, 3.0, 2.8, 2.3, 2.7, 2.5])
diet2 = np.array([3.0, 3.2, 3.1, 2.9, 2.8, 3.0, 3.2])

t_statistic, p_value = stats.ttest_ind(diet1, diet2)

print(f"t-stat: {t_statistic}")
print(f"p-value: {p_value}")
print(f"\nMean Diet 1: {np.mean(diet1):.2f} kg")
print(f"Mean Diet 2: {np.mean(diet2):.2f} kg")


t-stat: -3.5383407969933938
p-value: 0.004083270191713912

Mean Diet 1: 2.54 kg
Mean Diet 2: 3.03 kg


An independent samples t-test was conducted to determine if there was a significant difference in weight loss between the two diets. The results showed that Diet 1 had a mean weight loss of 2.54 kg, while Diet 2 had a mean weight loss of 3.03 kg.

The t-test yielded a t-statistic of -3.54 with a p-value of 0.004. Since the p-value (0.004) is less than the significance level of α = 0.05, we reject the null hypothesis.

Conclusion: There is statistically significant evidence that Diet 2 produces greater weight loss than Diet 1. On average, participants on Diet 2 lost approximately 0.49 kg more than those on Diet 1 over the one-month period.

## ANOVA

**ANOVA** (*Analysis of Variance*) is a statistical technique used to compare the measures of two or more groups. The idea behind ANOVA is to decompose the total variability in the data into two components: between-group variability and within-group variability:

- **Between-group variability**: This variability refers to the differences between the group means. If this variability is considerably larger than the within-group variability, it could be an indication that at least one of the group means is different.
- **Within-group variability**: This variability refers to the dispersion of the data within each group. If all groups have similar variability, then any noticeable difference in group means could be considered significant.

Hypotheses in ANOVA typically include:

- **Null hypothesis** ($H₀$): The means of all groups are equal.
- **Alternative hypothesis** ($H₁$): At least one of the group means is different.

If the ANOVA test result is significant (e.g., a p-value less than a threshold such as 0.05), this suggests that at least one group mean is different.

## Exercise 2

A farmer decides to test three different types of fertilizers to determine if one is superior in terms of corn production. The farmer plants corn on 15 identical plots and uses all three fertilizers (5 plots for each type). At the end of the season, he measures the corn yield (in kg) of each plot, with the following result:

| Fertilizer 1 | Fertilizer 2 | Fertilizer 3 |
|:-------------|:-------------|:-------------|
| 20 | 22 | 24 |
| 21 | 21 | 23 |
| 20 | 23 | 22 |
| 19 | 22 | 23 |
| 20 | 21 | 24 |

With this data, he seeks to answer the following question: Is there a significant difference in average corn yield between the three types of fertilizers?

To help you, follow the points below:

- State the hypothesis: null and alternative hypothesis.
- Perform the ANOVA test.
- Analyze the conclusions.
- If one fertilizer is better than another, how can we know it?

Null Hypothesis: There is no significant difference in average corn yield among the three types of fertilizers.H₁ 

Alternative Hypothesis: At least one fertilizer produces a significantly different average corn yield compared to the others.

In [3]:
# TODO
fertilizer1 = np.array([20, 21, 20, 19, 20])
fertilizer2 = np.array([22, 21, 23, 22, 21])
fertilizer3 = np.array([24, 23, 22, 23, 24])

f_statistic, p_value = stats.f_oneway(fertilizer1, fertilizer2, fertilizer3)

print(f"F-statistic: {f_statistic}")
print(f"p-value: {p_value}")
print(f"\nMean Fertilizer 1: {np.mean(fertilizer1):.2f} kg")
print(f"Mean Fertilizer 2: {np.mean(fertilizer2):.2f} kg")
print(f"Mean Fertilizer 3: {np.mean(fertilizer3):.2f} kg")

F-statistic: 20.31578947368421
p-value: 0.00014047824793190402

Mean Fertilizer 1: 20.00 kg
Mean Fertilizer 2: 21.80 kg
Mean Fertilizer 3: 23.20 kg


A one-way ANOVA test was performed to compare corn yield across three fertilizer types. The results showed significant differences among the groups (F = 20.32, p = 0.0001). Since the p-value (0.0001) is much less than the significance level of α = 0.05, we reject the null hypothesis.
The mean yields were:

Fertilizer 1: 20.00 kg
Fertilizer 2: 21.80 kg
Fertilizer 3: 23.20 kg

Conclusion: There is statistically significant evidence that at least one fertilizer performs differently than the others in terms of corn production.

Looking at the mean yields, Fertilizer 3 (23.20 kg) outperformed both Fertilizer 2 (21.80 kg) and Fertilizer 1 (20.00 kg). Given the highly significant ANOVA result (p = 0.0001) and the clear separation in means, Fertilizer 3 is the best choice for maximizing corn production, producing 3.2 kg more per plot than Fertilizer 1 and 1.4 kg more than Fertilizer 2.